### DISCUSSION TITLE
Support for ES modules

### DISCUSSION THREAD

**Discussion by chringwer** [artilleryio/artillery#1384] [Artillery]
Are there plans to support ES6 modules as processors or plugins for artillery? If we would use `await import()` instead of `require()`, I think that should work for both commonjs as well as modules. Of course that means we would need to deal with the asynchronicity here but that should be feasible. At least that's much less hassle than dealing with the async imports inside the processor function when dealing with a codebase that has migrated to `type: "module"`.

**Comment by hassy** [comment]
@chringwer yes, we're definitely going to add support for those! we just need to do it in a way that's fully backwards-compatible as there's a lot of code out there relying on current behavior.

**Reply by hassy** [reply]
@chringwer just landed on `main`, will be out in v2.0.7 on Monday. All hooks like `beforeRequest`, `afterResponse` etc can be `async` too now.
